packages/vite/src/node/server/transformRequest.ts

```diff
@@ -46,6 +47,8 @@ export function transformRequest(
   server: ViteDevServer,
   options: TransformOptions = {},
 ): Promise<TransformResult | null> {
+  if (server._restartPromise) throwClosedServerError()
```

Comment:
I wonder if it's worth having a method like `server.throwIfClosed()` to encapsulate this logic and potentially avoid needing to access a private variable. not sure if you'd want to expose this externally, but if so then middlewares could call it if necessary